generators/generators-commons/src/main/java/uk/gov/justice/services/generators/commons/helper/MessagingClientBaseUri.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.justice.services.generators.commons.helper;
+
+import static uk.gov.justice.services.generators.commons.helper.Names.buildJavaFriendlyName;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Parses RAML base URI for messaging client generation and exposes it's parts through accessor methods
+ */
+public class MessagingClientBaseUri {
+
+    private static final Pattern MESSAGING_BASE_URI_PATTERN = Pattern.compile("message://(.+)");
+    private final String uriExcludingProtocol;
+
+    public MessagingClientBaseUri(final String uriString) {
+        final Matcher m = matcherOf(uriString);
+        m.find();
+        this.uriExcludingProtocol = m.group(1);
```

Comment:
No it doesn't, as per the original. We need to refactor this completely to avoid making assumptions about the base URI at all.